### PR TITLE
Verification fixes

### LINKED ISSUE #313: continuous loading indicator is shown on Second & third fields when user auto paste attestations.
**Frequency:** 100%
**Repro on build version:** Android internal build V1.13.1(1004294349) 
**Repro on devices:** One plus 6T (10),Samsung Galaxy Note 5(7) , Redmi Note 8 (9.0)
**Pre-Condition:** Make user codes should paste automatically
   **Repro Steps:**

1. Launch the application 
2. Complete on boarding education
3. Tap on create new account 
4. Enter pin
5. Enter phone number 
6. Tap on start button 
7. Wait for phone number confirmation fields 
8. Make sure codes should auto paste 
9. Observe continuous loading indicator is shown on Second & third fields 

**Current Behavior:**  Continuous loading indicator is shown 
**Expected Behavior:**  Codes should auto paste & user should redirect to connect contacts screen. 
**Observation:** Codes are not auto pasted in ios. 
**Impact:** User  blocked to confirm phone number . 
**Attachment:**  [ContinuousLoadingIndicator.mp4](https://drive.google.com/file/d/1MmfuDGEXoMju7xSb8gFAvazWE-kwBFWG/view?usp=sharing)

### LINKED ISSUE #357: Amount not credited to user B’s account after completing signup & verification flow with same phone number which is invited by sending amount
**Frequency:** 100%
**Repro on build version:** Android internal build V1.14.0 (1004294350) , Test Flight build V1.14.0(58)
**Does not repro version**  Play Store build V1.12.2, App store build V1.12.2
**Repro on devices:** iPhone SE (14.2), Redmi Note 8 (9.0), iPhone 6+(12.4.5)
**Testing Account:**  (680) 333 1978 / [Backup Key](https://docs.google.com/spreadsheets/d/1gXY1I4PBkT9f1uB3zxWgUUJJ1zjVeJcqcWLjiIls1f0/edit#gid=0&range=B9) 
**Pre-condition:** Users should be invited with a link  to any phone number with some amount.
 **Repro Steps:**
1. Launch the invited link URL in user B’s device.
2. Complete Signup flow.
3. Enter same number which was invited through link
4. Complete verification flow & redirect to Home screen
5. Now wait for some moment & observed that,
**Current Behavior:**  Account balance not credited to user B’s account.
**Expected Behavior:**   Account balance should be credited to user B’s account & respective Activity should be displayed in “Recent” feed.
**Impact:** User not able to get amount which sent through invitation
**Attachment:**  [AmountNotCreditedAfterLogin&VerifyingThroughInvitation.mp4](https://drive.google.com/file/d/1LMoRur72h_4BbABb67r6EhcWIVjJJw2v/view?usp=sharing)

### PR DESCRIPTION
**Pull request by gnardini**
### Description

- Set the MTW during onboarding so that it can be used to reclaim escrow (and other things probably).
- Make a delay a bit larger to avoid race condition.
- Avoid calling the verification saga more than once by not calling `fetchOnChainData` when an attestation is completed.
- Make sure a lock is released when an error happens.
- Add more logs in general.

### Other changes

N/A

### Tested

Manually, I invited myself and reclaimed a few times. I temporarily changed the code of the send flow a bit to allow me to invite my phone number even though I already have an account.

### How others should test

- Invite a new phone number to join Valora
- Redeem the invite and reclaim the funds.

### Related issues

- Fixes #357
- Fixes #313

### Backwards compatibility

N/A

### CODE REVIEW

**Review by jeanregisser — COMMENTED** [review]
Thanks!
From what I see from the verification flow prior to the refactor, `setMtwAddress` was only called once fully verified.

https://github.com/celo-org/wallet/blob/7ec2a50151c643183166531e90d7ac5213fe7227/packages/mobile/src/identity/feelessVerification.ts#L1139-L1153

Is there a reason why we're not doing the same here?

**Comment by gnardini** [comment]
I'm not sure, I guess not? If a user already has a `mtwAddress` even if it's not verified we should use that though, right? Deploying a new one uses up gas so no reason to throw the MTW away if it wasn't completely verified. 

Not sure if the code handles having a not-verified MTW though, I'll take a deeper look to see if it makes sense

**Comment by MuckT** [comment]
Verified the fix for #313 on an Pixel 4 on Android 10 (API 29).

**Review by MuckT — APPROVED** [review]
Confirmed fix for #313.

File: packages/mobile/src/identity/securityCode.ts [hunk 1]
```diff
@@ -78,15 +78,16 @@ async function requestValidator(
...
       attestation.attestationServiceURL,
       requestBody,
       signer
     )
+    return attestationCode
   } catch (error) {
     Logger.error(
       TAG + '@getAttestationCodeFromSecurityCode',
-      `get for issuer ${issuer} failed`,
+      `get for issuer ${issuer} failed - ${JSON.stringify(error)}`,
```
Comment (tarikbellamine) [inline packages/mobile/src/identity/securityCode.ts:90]: Don't think this change is necessary anymore. `Logger.error` should be handing `error` correctly now

File: packages/mobile/src/verify/saga.ts [hunk 2]
```diff
@@ -689,6 +688,4 @@ export function* verifySaga() {
   yield takeEvery(fail.type, failSaga)
   yield takeEvery(reset.type, resetSaga)
   yield takeEvery(stop.type, stopSaga)
-  // TODO: this can be calculated in reducer, once we stop using identify/reducer for verification
-  yield takeEvery(Actions.COMPLETE_ATTESTATION_CODE, fetchOnChainDataSaga)
```
Comment (jeanregisser) [inline packages/mobile/src/verify/saga.ts:693]: I śaw your comment about this, are we sure there's not side effect to removing this?

**Review by jeanregisser — APPROVED** [review]
Amazing!! 🚀

**Comment by MuckT** [comment]
https://valora.testquality.com requirement [R21](https://valora.testquality.com/project/12128/testcase/282113/requirement/11275). Do not remove or edit.

**Comment by Celoqa** [comment]
Hey @gnardini  I have verified this issue using latest released Android internal build & Test Flight Build V1.14.2 & observed following:
**Observations:**

- [x] User B is able to reclaim the funds & also able to see reclaimed amount in Hamburger Menu after Redeeming the invite.

- [x] User B also able to see  "Dollars transferred to your account!" toast message after reclaiming amount.

- [x] Also it is observed that, both "Payment Received You've Received $# Celo Dollars" In-app CTA & activity displayed in "Recent" feed. [Attachment](https://drive.google.com/file/d/10I3r2WpgxkBzdNzeAXPwOj_Gd8WHtrSB/view?usp=sharing)

- [ ] User is able to see loading indicator for short time while pasting second code automatically. Also it is observed that user stays on paste attestation codes (for ~10 sec) even pasted all codes successfully. Can you please let us know is this expected or we need to raise a ticket for same ? [Attachement](https://drive.google.com/file/d/1gDdrvkia78gotDzc1kvW49Xq3-t5eDOk/view?usp=sharing)

Thanks..!!

**Comment by gnardini** [comment]
Hi @Celoqa 
Thanks for the report. The issue with the video you sent is not the 10 second delay but the fact that there's a repeated code so verification wasn't really successful even though you got a success screen.

I created an issue here: https://github.com/celo-org/wallet/issues/390

Thanks for catching this!

**Comment by Celoqa** [comment]
Hi @gnardini thanks for the update.
We will verify  #390 on latest build Android internal V1.14.2 and will update the status in ticket.

**Pull request merged**